stan/math/rev/fun/dot_product.hpp

```diff
@@ -38,22 +39,27 @@ inline return_type_t<T1, T2> dot_product(const T1& v1, const T2& v2) {
   const auto& v1_col = as_column_vector_or_scalar(v1);
   const auto& v2_col = as_column_vector_or_scalar(v2);
 
-  const auto& v1_val = to_arena_if<!is_constant<T2>::value>(value_of(v1_col));
-  const auto& v2_val = to_arena_if<!is_constant<T1>::value>(value_of(v2_col));
+  const auto& v1_val_arena
+      = to_arena_if<!is_constant<T2>::value>(value_of(v1_col));
```

Comment:
Any particular reason to switch from `auto` to `const auto&` here?

Reply:
Actually the type here could be `arena_t<promote_scalar_t<double, T1>>` which makes the type explicit which is nice and comforting.